test-clients/src/main/java/com/hedera/services/bdd/spec/transactions/TxnVerbs.java

```diff
@@ -330,6 +353,105 @@ public static HapiContractUpdate contractUpdate(String contract) {
+		final var path = getResourcePath(contractName, ".bin");
+		final var file = new HapiFileCreate(contractName);
+		sourcing(() -> file).execFor(spec);
+		sourcing(() -> updateLargeFile(GENESIS, contractName, extractByteCode(path))).execFor(spec);
+		return file;
+	}
+
+	/*  Note to the reviewer:
+		This method is temporarily named with the "new" prefix, as soon as the implementation is approved and the legacy
+		contractCreate() is entirely replaced, this method will be renamed to "contractCreate"
+	 */
+	public static HapiContractCreate newContractCreate(final String contractName, final Object... constructorParams) {
+		if (constructorParams.length > 0) {
+			final var constructorABI = getABIFor(CONSTRUCTOR, EMPTY, contractName);
+			return new HapiContractCreate(contractName, constructorABI, constructorParams).bytecode(contractName);
+		} else {
+			return new HapiContractCreate(contractName).bytecode(contractName);
+		}
+	}
+
+	/**  Note:
+	 Previously - when creating contracts we were passing a name, chosen by the developer, which can differentiate
+	 from the name of the contract.
+	 The new implementation of the contract creation depends on the exact name of the contract, which means, that we
+	 can not create multiple instances of the contract with the same name.
+	 Therefore - in order to provide each contract with a unique name - the developer must attach a suffix to happily
+	 create multiple instances of the same contract, but with different names.
+	 Example:
+	 final var contract = "TransferringContract";
+	 newContractCreate(contract).balance(10_000L).payingWith(ACCOUNT),
+	 newContractCreate(contract, to).balance(10_000L).payingWith(ACCOUNT)
+	 */
+	public static HapiContractCreate cloneContract(final String contractName, final String suffix, final Object... constructorParams) {
+		if (constructorParams.length > 0) {
+			final var constructorABI = getABIFor(CONSTRUCTOR, EMPTY, contractName);
+			return new HapiContractCreate(contractName + suffix, constructorABI, constructorParams).bytecode(contractName);
+		} else {
+			return new HapiContractCreate(contractName + suffix).bytecode(contractName);
+		}
+	}
+
+	/**This method enables the developer to create a file and deploy a contract with a single method call.
+	 The execution of the method requires the op context, therefore the invocation is not straightforward:
+	 "withOpContext((spec, log) -> contractDeploy(contractName, spec).execFor(spec));"
+	 It can be convenient when the developer deploys a nested contract, since both the creation of the outer and the inner contract
+	 can happen in the given clause:
+	 "withOpContext((spec, log) -> contractDeploy(INNER_CONTRACT, spec).execFor(spec)),
+	 withOpContext((spec, log) -> contractDeploy(OUTER_CONTRACT, spec, getNestedContractAddress(INNER_CONTRACT, spec)).execFor(spec))"
+	 */
+	public static HapiContractCreate contractDeploy(final String contractName,
```

Comment:
This method is unused and we should remove it.

Reply:
Solved